app/Platform/Commands/SyncGameSets.php

```diff
@@ -4,35 +4,91 @@
-use Exception;
+use App\Models\Game;
+use App\Models\GameAlternative;
+use App\Models\GameSet;
+use App\Models\GameSetGame;
+use App\Models\GameSetLink;
+use App\Models\System;
+use App\Platform\Enums\GameSetType;
 use Illuminate\Console\Command;
 
 class SyncGameSets extends Command
 {
-    use SyncTrait;
-
     protected $signature = 'ra:sync:game-sets {--f|full} {--p|no-post}';
-    protected $description = 'Sync game alternatives to sets';
+    protected $description = 'Sync hubs to game_sets';
 
     public function __construct()
     {
         parent::__construct();
     }
 
-    /**
-     * @throws Exception
-     */
     public function handle(): void
     {
-        $this->sync('game_sets');
-    }
+        $this->info('Running a full GameAlternatives -> game_sets sync...');
+
+        // This will be a full reset. Delete any existing game_sets data.
+        $this->info("\nDeleting any existing game_sets data...");
+        GameSetLink::query()->forceDelete();
+        GameSetGame::query()->forceDelete();
+        GameSet::query()->forceDelete();
+        $this->info("Deleted all existing game_sets data.");
+
+        // Loop through all GameAlternatives and create game_sets.
+        $distinctGameIds = GameAlternative::select('gameID')->distinct()->pluck('gameID');
+        $distinctGameIdsCount = $distinctGameIds->count();
+
+        $this->info("\nUpserting {$distinctGameIdsCount} game_sets derived from legacy GameAlternatives.");
+        $progressBar = $this->output->createProgressBar($distinctGameIdsCount);
+        foreach ($distinctGameIds as $gameId) {
+            $game = Game::find($gameId);
+
+            if ($game->ConsoleID === System::Hubs) {
+                GameSet::updateOrCreate(
+                    ['game_id' => $game->id],
+                    ['title' => $game->title, 'type' => GameSetType::HUB],
+                );
+            } else {
+                GameSet::updateOrCreate(
+                    ['game_id' => $game->id],
+                    ['type' => GameSetType::GAME]
+                );
+            }
 
-    // protected function preProcessEntity(object $origin, array $transformed): array
-    // {
-    // }
+            $progressBar->advance();
+        }
+        $progressBar->finish();
 
-    // protected function postProcessEntity(object $origin, object $transformed): void
-    // {
-    // }
+        $gameAltsCount = GameAlternative::count();
+        $this->info("\nPopulating {$gameAltsCount} game_set_games and game_set_links...");
+
+        $progressBar = $this->output->createProgressBar($gameAltsCount);
+
+        foreach (GameAlternative::cursor() as $gameAlt) {
+            $parentGameSet = GameSet::firstWhere('game_id', $gameAlt->gameID);
+            $game = Game::find($gameAlt->gameID);
+            $altGame = Game::find($gameAlt->gameIDAlt);
+
+            // Only process relationships in one direction so we avoid duplicate game_set_games and game_set_links.
```

Comment:
This doesn't maintain the bi-directionality of GameAlternatives.

For example, Dragon Quest I (1471) is similar to Hydlide (1742). Currently, there's two rows in GameAlternatives for this:
```
+--------+-----------+---------+---------------------+
| gameID | gameIDAlt | Created | Updated             |
+--------+-----------+---------+---------------------+
|   1471 |      1742 | NULL    | 2019-09-22 20:58:58 |
|   1742 |      1471 | NULL    | 2019-09-22 20:58:58 |
+--------+-----------+---------+---------------------+
```
But after running this code, the game set for game 1471 contains the 1742 game, but the game set for game 1742 does not contain the 1471 game.

Reply:
I don't think we want bi-directionality for games in hubs.

I also don't think we want bi-directionality for hubs in hubs, but I'm not sure how easy it would be to determine the parent chain for something like:

`Central` > `Central - Genre & Subgenre` > `Genre - Board and Card` > `Subgenre - Card Game` > `Subgenre - Poker`

Also, some hubs have related hubs, like `Subgenre - Tactical RPG` and `Subgenre - Turn-based RPG`.

Reply:
With the changes in latest:
* Bi-directionality should be supported for "non-hub" games.
* Bi-directionality should be guarded against for "hub" games. The relations are in one direction.

I've confirmed that with the latest changes, bi-directionality is intact for 1471<->1742.